tbdex/rfq/rfq.go

```diff
@@ -34,8 +35,8 @@ func (r *RFQ) Sign(bearerDID did.BearerDID) error {
 	return nil
 }
 
-// UnmarshalJSON validates and unmarshals the input data into an RFQ.
-func (r *RFQ) UnmarshalJSON(data []byte) error {
+// ValidateAndUnmarshalJSON validates and unmarshals the input data into an RFQ.
+func (r *RFQ) ValidateAndUnmarshalJSON(data []byte, verifyPrivateDataStrict bool) error {
```

Comment:
low key i prefer `UnmarshallJSON`. i asked chad and he thinks it's okay to have custom validations in an `UnmarshallJSON` method.

Reply:
Agreed minus the additional `l` in `Unmarshal` bc go interface

Reply:
@mistermoe what's the suggestion here? `UnmarshalJSON`? 

Changed it because I was getting an error that a method named `UnmarshalJSON` should have the signature `UnmarshalJSON(data []byte)` without the bool.

Reply:
Aye bc https://pkg.go.dev/encoding/json#Unmarshaler

We could scrap the bool and just verify `privateData` if it's not nil

Reply:
Yea that would fix the error, do you know why we needed the bool here? https://github.com/TBD54566975/tbdex-js/blob/main/packages/protocol/src/message-kinds/rfq.ts#L78-L82